File: test-optional-lib.js
/**
 * Library with functions that accept optional parameters.
 * Used for testing optional param type generation.
 */

function greet(name, greeting) {
  if (greeting) {
    return `${greeting}, ${name}!`;
  }
  return `Hello, ${name}!`;
}

function search(query, limit, offset) {
  const results = ['result1', 'result2', 'result3'];
  const start = offset || 0;
  const end = (limit || 10) + start;
  return { results: results.slice(start, end), query, total: results.length };
}

class Config {
  get(key, defaultValue) {
    const store = { theme: 'dark', lang: 'en' };
    return store[key] || defaultValue || null;
  }
}

module.exports = { greet, search, Config };

File: test_optional_lib.py
"""Library with functions that accept optional parameters."""


def greet(name, greeting=None):
    if greeting:
        return f"{greeting}, {name}!"
    return f"Hello, {name}!"


def search(query, limit=None, offset=None):
    results = ["result1", "result2", "result3"]
    start = offset or 0
    end = (limit or 10) + start
    return {"results": results[start:end], "query": query, "total": len(results)}


class Config:
    def get(self, key, default_value=None):
        store = {"theme": "dark", "lang": "en"}
        return store.get(key) or default_value
